### PR TITLE
Add method for creating solid objects from list of boundaries and filled regions.

### LINKED ISSUE #1121: Add method for creating solid objects from list of boundaries and filled regions.
#### Description:

As in the title.

### PR DESCRIPTION
**Pull request by michal-pekacki**
### Issues addressed by this PR

Closes #1121

Add method for creating solid extrusion from `IList<CurveLoop>` and `Filled Region` with bottom and top limitation.

### CODE REVIEW

File: Revit_Core_Engine/Create/Solid.cs [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+                BH.Engine.Reflection.Compute.RecordError($"Filled region cannot be null.");
+                return null;
+            }
+            else if (topElevation - bottomElevation < Tolerance.Distance)
+            {
+                BH.Engine.Reflection.Compute.RecordError($"Top elevation value must be greater than bottom elevation.");
+                return null;
+            }
+
+            Document doc = filledRegion.Document;
```
Comment (pawelbaran) [inline Revit_Core_Engine/Create/Solid.cs:34]: `doc` is redundant here from what I can see

File: Revit_Core_Engine/Create/Solid.cs [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+            }
+            else if (topElevation - bottomElevation < Tolerance.Distance)
+            {
+                BH.Engine.Reflection.Compute.RecordError($"Top elevation value must be greater than bottom elevation.");
+                return null;
+            }
+
+            double elev = boundaries[0].GetPlane().Origin.Z;
+            XYZ dir = XYZ.BasisZ;
+            double height = topElevation - bottomElevation;
```
Comment (pawelbaran) [inline Revit_Core_Engine/Create/Solid.cs:62]: Something tells me that `topElevation` and `bottomElevation` should be converted `FromSI` before being used to call Revit API methods - please remember that the standard unit system used by BHoM is meters, not Revit-beloved feet, knees and ankles 😉

File: Revit_Core_Engine/Create/Solid.cs [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+            Solid solid = GeometryCreationUtilities.CreateExtrusionGeometry(boundaries, dir, height);
+
+            if (Math.Abs(elev - bottomElevation) > Tolerance.Distance)
+            {
+                XYZ translation = new XYZ(0, 0, bottomElevation - elev);
+                Transform transform = Transform.CreateTranslation(translation);
+                Solid solid2 = SolidUtils.CreateTransformed(solid, transform);
+                solid = solid2;
```
Comment (pawelbaran) [inline Revit_Core_Engine/Create/Solid.cs:70]: Not a mistake, but I think overwriting the variable in one line would be cleaner here
```suggestion
                solid  = SolidUtils.CreateTransformed(solid, transform);
```

**Review by pawelbaran — CHANGES_REQUESTED** [review]
I really like the overall idea. However, I think there would be an issue with units meters vs feet - please see the comments. Besides that all good!

**Comment by pawelbaran** [comment]
@BHoMBot check compliance
@BHoMBot check null-handling

File: Revit_Core_Engine/Create/Solid.cs [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+        /***************************************************/
+        /****              Public methods               ****/
+        /***************************************************/
+
+        [Description("Create solid object from filled region limited by bottom and top elevation.")]
+        [Input("filledRegion", "Filled region that is extruded to the solid.")]
+        [Input("bottomElevation", "Bottom elevation of the solid.")]
+        [Input("topElevation", "Top elevation of the solid.")]
+        [Output("solid", "Solid representation of extruded filled region.")]
+        public static Solid Solid(this FilledRegion filledRegion, double bottomElevation, double topElevation)
```
Comment (pawelbaran) [inline Revit_Core_Engine/Create/Solid.cs:21]: I think the general approach is not to use `this` keyword in `Create` methods - these methods should remind factory pattern.
```suggestion
        public static Solid Solid(FilledRegion filledRegion, double bottomElevation, double topElevation)
```

File: Revit_Core_Engine/Create/Solid.cs [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+        }
+
+        /***************************************************/
+
+        [Description("Create solid object from boundaries limited by bottom and top elevation.")]
+        [Input("boundaries", "Boundaries list that is extruded to the solid.")]
+        [Input("bottomElevation", "Bottom elevation of the solid.")]
+        [Input("topElevation", "Top elevation of the solid.")]
+        [Output("solid", "Solid representation of extruded boundaries.")]
+        public static Solid Solid(this IList<CurveLoop> boundaries, double bottomElevation, double topElevation)
```
Comment (pawelbaran) [inline Revit_Core_Engine/Create/Solid.cs:47]: ...and same as above
```suggestion
        public static Solid Solid(IList<CurveLoop> boundaries, double bottomElevation, double topElevation)
```

**Review by pawelbaran — CHANGES_REQUESTED** [review]
Two more minor changes pls

**Comment by pawelbaran** [comment]
@BHoMBot check compliance
@BHoMBot check null-handling

**Comment by pawelbaran** [comment]
The code looks good, but you still need to add the copyright to the file as well as fix project compliance (each BHoM assembly reference in the .csproj file needs to have `Private` and `SpecificVersion` attributes set to `false`.

**Comment by michal-pekacki** [comment]
@BHoMBot fix project file ref. 4210047549

**Comment by michal-pekacki** [comment]
@BHoMBot check compliance

**Comment by pawelbaran** [comment]
@BHoMBot check null-handling

**Review by pawelbaran — APPROVED** [review]
LGTM

**Comment by pawelbaran** [comment]
@BHoMBot check core
@BHoMBot check installer
@BHoMBot check serialisation
@BHoMBot check versioning

**Comment by IsakNaslundBh** [comment]
@BHoMBot this is a CI/CD instruction. I am authorising dispensation to be granted on check ref. 4211967657

**Comment by pawelbaran** [comment]
@BHoMBot check ready-to-merge

**Pull request merged**